include/envoy/router/router.h

```diff
@@ -1174,10 +1189,12 @@ class GenericConnectionPoolCallbacks {
                            const Network::Address::InstanceConstSharedPtr& upstream_local_address,
                            const StreamInfo::StreamInfo& info) PURE;
 
-  // TODO(alyssawilk) This exists because the Connection Pool creates the GenericUpstream, and the
-  // GenericUpstream needs a handle back to the upstream request to pass on events, as upstream
-  // data flows in. Do interface clean up in a follow-up PR.
-  virtual UpstreamRequest* upstreamRequest() PURE;
+  // @return the UpstreamToDownstream interface for this stream.
+  //
+  // This is the interface for all interactions the upstream stream needs to have with the
+  // downstream stream. It is in the GenericConnectionPoolCallbacks as the GenericConnectionPool
+  // creates the GenericUpstream, and the GenericUpstream will need this interface.
+  virtual UpstreamToDownstream* upstreamToDownstream() PURE;
```

Comment:
Can this return a reference instead of pointer? (Alternate formation of the same question: when can this be null?)